authentication/account/service/user.go

```diff
@@ -0,0 +1,127 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/fabric8-services/fabric8-auth/application/service"
+	"github.com/fabric8-services/fabric8-auth/authorization/token/manager"
+
+	"github.com/fabric8-services/fabric8-auth/application/service/base"
+	servicecontext "github.com/fabric8-services/fabric8-auth/application/service/context"
+	"github.com/fabric8-services/fabric8-auth/authentication/account/repository"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/satori/go.uuid"
+)
+
+// NewUserService creates a new service to manage users
+func NewUserService(ctx servicecontext.ServiceContext) service.UserService {
+	return &userServiceImpl{
+		BaseService: base.NewBaseService(ctx),
+	}
+}
+
+// userServiceImpl implements the UserService to manage users
+type userServiceImpl struct {
+	base.BaseService
+}
+
+// UserInfo gets user information given a context containing access_token
+func (s *userServiceImpl) UserInfo(ctx context.Context, identityID uuid.UUID) (*repository.User, *repository.Identity, error) {
+	var identity *repository.Identity
+	err := s.ExecuteInTransaction(func() error {
+		var err error
+		identity, err = s.Repositories().Identities().LoadWithUser(ctx, identityID)
+		if err != nil || identity == nil {
+			return errors.NewUnauthorizedError(fmt.Sprintf("auth token contains id %s of unknown Identity\n", identityID))
+		}
+		return nil
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+	log.Debug(ctx, map[string]interface{}{
+		"identity_id": identity.ID,
+		"user_id":     identity.User.ID,
+	}, "loaded identity and user")
+	return &identity.User, identity, nil
```

Comment:
why not simply returning `identity, nil` and let the caller function get the associated `user` from the returned `identity`?

Reply:
I'm not sure why this function was written this way, and while I agree with you that it would be simpler to return identity I'd prefer to refactor this in a separate issue in the interest of limiting any further changes to this PR.